### ISSUE TITLE
Always need to connect to elasticsearch?

### ISSUE THREAD

**Issue by ramsaylee** [iverberk/larasearch#51]
Hi,

First of all, I want to thank you for this great package. I am new to Laravel and Elasticsearch, it helps me a lot.

It works great with my app, but I got the error below when I am not connecting to elasticsearch service, could you please help?

{"error":{"type":"Elasticsearch\\Common\\Exceptions\\Curl\\CouldNotConnectToHost","message":"Failed to connect to localhost port 9200: Connection refused","file":"Myapplication Root Directory\\vendor\\elasticsearch\\elasticsearch\\src\\Elasticsearch\\Connections\\AbstractConnection.php","line":308}} 

I know the error happended because my app cannot connect to elasticsearch service, but I do not want to connect to elasticsearch when login or other actions (no need to connect to elasticsearch), I only want to connect to elasticsearch when search. Could you please tell me is there anything I can do this?

I come from China, sorry for my poor English, if you do not know what I mean exactly, please let me know.

Thanks a lot!

Happy New Year!

Lee

**Comment by iverberk** [comment]
Hi Lee,

Thanks for your kind words. Your English is fine :-)

The connection to Elasticsearch is probably due to the automatic model (re)indexing. Did you include the searchable-/callabletrait on the models that are used during login or other actions?

You can prevent the reindex by defining a custom shouldIndex method on your model. See the documentation [here](https://github.com/iverberk/larasearch/wiki/Automatic-reindexing)

Hope this help, let me know.

**Comment by ramsaylee** [comment]
Hi iverberk,

Thanks for the quick and helpful answer.

Yes, I included the searchabletrait on the model usesd during login. I should have read the wiki more carefully at first. I will try the "custom shouldIndex method" solution as you adviced and I believe the problem will be resolved.

I will close this issue.

Again, thanks for your helpful answer and this great package.

Lee

**Comment by ramsaylee** [comment]
Hi Iverberk,

Sorry to interrupt you again. I have to reopen this issue.

I am back to work on my project yesterday, and I tried the "custom shouldIndex method" solution as you adviced 10 days ago, but it doesn't work. 

I have added shouldIndex method on all the models have SearchableTrait/CallableTrait, and then do some actions(like login), Laravel still tells me that "cannot connect to elasticsearch".

Did I do it in the wrong way?

Thanks in advance!

Lee
